statsprinter.go

```diff
@@ -159,44 +213,83 @@ func (p *planePrinter) printStatistics(t tcping) {
 }
 
 func (p *planePrinter) printProbeSuccess(hostname, ip string, port uint16, streak uint, rtt float32) {
+	var message string
 	if hostname == "" {
-		colorLightGreen("Reply from %s on port %d TCP_conn=%d time=%.3f ms\n",
-			ip, port, streak, rtt)
-		return
+		message = fmt.Sprintf("Reply from %s on port %d TCP_conn=%d\n", ip, port, streak)
+	} else {
+		message = fmt.Sprintf("Reply from %s (%s) on port %d TCP_conn=%d\n", hostname, ip, port, streak)
 	}
-
-	colorLightGreen("Reply from %s (%s) on port %d TCP_conn=%d time=%.3f ms\n",
-		hostname, ip, port, streak, rtt)
+	options := &PrintOptions{
+		Color:         Green,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printProbeFail(hostname, ip string, port uint16, streak uint) {
+	var message string
 	if hostname == "" {
-		colorRed("No reply from %s on port %d TCP_conn=%d\n",
-			ip, port, streak)
-		return
+		message = fmt.Sprintf("No reply from %s on port %d TCP_conn=%d\n", ip, port, streak)
+	} else {
+		message = fmt.Sprintf("No reply from %s (%s) on port %d TCP_conn=%d\n", hostname, ip, port, streak)
+	}
+	options := &PrintOptions{
+		Color:         Red,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
 	}
-	colorRed("No reply from %s (%s) on port %d TCP_conn=%d\n",
-		hostname, ip, port, streak)
+	printReply(options)
 }
 
 func (p *planePrinter) printTotalDownTime(downtime time.Duration) {
-	colorYellow("No response received for %s\n", durationToString(downtime))
+	message := fmt.Sprintf("No response received for %s\n", durationToString(downtime))
+	options := &PrintOptions{
+		Color:         Yellow,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printRetryingToResolve(hostname string) {
-	colorLightYellow("retrying to resolve %s\n", hostname)
+	message := fmt.Sprintf("retrying to resolve %s\n", hostname)
+	options := &PrintOptions{
+		Color:         LightYellow,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printInfo(format string, args ...any) {
-	colorLightBlue(format+"\n", args...)
+	message := fmt.Sprintf(format+"\n", args...)
+	options := &PrintOptions{
+		Color:         LightCyan,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printError(format string, args ...any) {
-	colorRed(format+"\n", args...)
+	message := fmt.Sprintf(format+"\n", args...)
+	options := &PrintOptions{
+		Color:         Red,
+		Message:       message,
+		ShowTimestamp: p.ShowTimestamp,
+	}
+	printReply(options)
```

Comment:
This does not need a timestamp

Reply:
could u please elaborate? 
We print the timestamp for succesful probes but not for unsuccesful probes?

Reply:
errors regarding probe failures use `printProbeFail`. The `printError` is used for reporting errors, usually before exiting. You can see the examples in tcping.go

Reply:
ahh got it, thank u